File: Valid Parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if s == '':
            return True
        elif len(s) % 2 == 1:
            return False

        brackets = {
            '(': {')': False, ']': True, '}': True},
            '[': {')': True, ']': False, '}': True},
            '{': {')': True, ']': True, '}': False},
        }

        stack = []
        for x in list(s):
            if x in ['(', '[', '{']:
                stack.append(x)
            elif not len(stack) or brackets[stack.pop()][x]:
                return False

        return not len(stack)


solution = Solution()
print(solution.isValid('()[]{}'))  # True
print(solution.isValid('([)]'))  # False

File: Valid Parentheses/valid-parentheses.js
/**
 * Check is valid parentheses
 *
 * @param {string} s
 * @return {boolean}
 */
const isValid = function(s) {

  const brackets = {
    '{': '}', '[': ']', '(': ')'
  };
  let stack = [];

  for (let i = 0; i < s.length; i++) {
    let cur = s[i];
    if (brackets[cur] !== undefined) {
      stack.push(brackets[cur]);
    } else if (stack.pop() !== cur) {
      return false;
    }
  }

  return !stack.length;
};

console.log(isValid('()[]{}')); // true
console.log(isValid('([)]')); // false